posts/2025-01-22-microprofile-telemetry-20.adoc

```diff
@@ -0,0 +1,325 @@
+
+This post demonstrates an application consisting of two services which are hosted on separate servers: 
+
+* *Person service*: Allows users to create records of people, given their name and age.
+
+* *Car service*: Allows users to create records of cars, given the make of the car and the id of the person who owns the car.
+
+image::img/blog/car_person_service.png[align="center",width=50%,Diagram of an application with a Car and Person service.]
+
+When a user interacts with the car service, they experience some latency with their response but the reason for this is not yet clear. There are numerous possibilities that could cause the latency such as network unreliability, hardware limitations, request timeouts and code complexity. The exact operation that has the problem is not always obvious. Therefore, we would need to collect data to work out which part of the distributed system is causing the issue.
+
+To do this, we can collect:
+
+* *Logs* from the servers and application to be aware of events and the time that they occurred.
+* *Runtime metrics* including numerical data about the infrastructure for the application, for example, CPU utilization and the JVM behaviour including the number of threads and classes.
+* *Application metrics* for the incoming and outgoing HTTP requests.
+* *Traces* and *spans* to track user operations, including both the requests and responses.
+
+Exporters are systems that send data to a backend service so you can visualise and monitor the spans, metrics and logs.
+
+The following section illustrates how to set up a backend service.
+
+== Running the monitoring backend
+
+The backend system is set up to collect the telemetry data from the application. See the link:https://github.com/grafana/docker-otel-lgtm[docker-otel-lgtm] repository and clone the repo:
+
+[source,sh]
+----
+git clone https://github.com/grafana/docker-otel-lgtm.git \
+cd docker-otel-lgtm/docker
+----
+
+Build and run the image containing all services required for gathering traces, metrics and logs:
+
+[source,sh]
+----
+podman build . -t grafana/otel-lgtm \
+podman run -p 3000:3000 -p 4317:4317 -p 4318:4318 --rm -ti localhost/grafana/otel-lgtm
+----
+
+The backend system is set up to collect telemetry data via the OTLP protocol which is default behaviour for Open Telemetry. Therefore, additional configuration for the exporter is not required in the application.
+
+image::img/blog/otel-lgtm-car-person.png[align="center",width=50%,Diagram of the otel lgtm backend exporter with the Car and Person service.]
+
+== The demo app
+
+Check out the demo repo and clone the repository: 
+
+git clone https://github.com/yasmin-aumeeruddy/car-person-demo.git
+
+Each service has `mpTelemetry2.0` and `monitor-1.0` enabled in its link:https://github.com/yasmin-aumeeruddy/car-person-demo/blob/main/car/src/main/liberty/config/server.xml#L8[server]:
+
+[source,xml]
```

Comment:
none of our existing docs for mptelemetry mention enabling monitor-1.0. If this is a requirement, or even a common use case, please open a doc bug that explains how we can provide this information in the docs